### ISSUE TITLE
its预训练模型

### ISSUE THREAD

**Issue by Wjc369** [deng-ai-lab/SFHformer#10]
您好，我用了您提供的its预训练模型测试sots-insdoor数据集时发现他的psnr只有24，和预期的43不符合，期待您的回复

**Comment by jxy33zrhd** [comment]
Hello 请确保用了正确的SOTS-indoor测试集 官方下载的SOTS-Indoor的gt部分有一些问题

![Image](https://github.com/user-attachments/assets/b8d70a7b-4619-4446-964f-56533e847a4e)

您可以看到这个gt的边缘有一些突出 它的尺寸是640\*480 但是inpu的图片 如下图 边缘是没有突出的 输入的尺寸是620\*460

![Image](https://github.com/user-attachments/assets/8eaf104d-f4ab-4e57-96c1-d84a9f2317d9)

因此 为了消除这一影响 我们将gt部分多出的20\*20部分进行了中心裁剪以确保一致 如下图

![Image](https://github.com/user-attachments/assets/4115a3dc-a23b-4232-abc4-048718ed3003)

裁剪后的这个数据可以在我们上传的SOTS数据集上面获取

我觉得大概率是上述原因导致的 您可以检查一下所用的SOTS-indoor数据集 如果不是上述原因 可否提供一下您输出的图像结果？以便更好地判断是什么原因造成的

**Comment by Wjc369** [comment]
您好，我仔细检查了发现不是数据集的问题，、但是当输出图片时尺寸变成了616*456

![Image](https://github.com/user-attachments/assets/7e434a4c-fa4d-4bf6-bebd-fac8228038f3)

**Comment by jxy33zrhd** [comment]
![Image](https://github.com/user-attachments/assets/7b0ea060-50ba-420c-b53d-14f41db0b676)

Hello 我在服务器上面重新测试了一遍 测试结果的PSNR为43.03
可否给我发一份您的测试代码呢 我帮您看一下

**Comment by Wjc369** [comment]
好的好的，麻烦了  
import os
os.environ['CUDA_VISIBLE_DEVICES'] = "0"
import torchvision.utils as vutils
import argparse
import json
import torch
import torch.nn as nn
import torch.nn.functional as F
from torch.utils.data import DataLoader, RandomSampler
from tensorboardX import SummaryWriter
from tqdm import tqdm
from utils import AverageMeter
from datasets.ITS_Dataloader import TrainData, ValData, TestData_its
from numpy import *
from pytorch_msssim import ssim
# from pytorch_ssim import SSIM
from models import *
from einops import rearrange
import torch.nn.functional as F
# from matplotlib import pyplot as plt

parser = argparse.ArgumentParser()
parser.add_argument('--model', default='sfhformer_haze', type=str, help='model name')
parser.add_argument('--num_workers', default=0, type=int, help='number of workers')
parser.add_argument('--save_dir', default='./saved_models/', type=str, help='path to models saving')
parser.add_argument('--data_dir', default='./data/', type=str, help='path to dataset')
parser.add_argument('--log_dir', default='./logs/', type=str, help='path to logs')
parser.add_argument('--exp', default='indoor', type=str, help='experiment setting')
args = parser.parse_args()

torch.manual_seed(8001)

def valid(val_loader448, network):
    os.makedirs('./result', exist_ok=True)

    PSNR448 = AverageMeter()
    SSIM_full = AverageMeter()
    torch.cuda.empty_cache()

    network.eval()


    for idx, batch in enumerate(val_loader448):
        source_img = batch['source'].cuda()
        target_img = batch['target'].cuda()
        _, _, H, W = target_img.shape

        with torch.no_grad():  # torch.no_grad() may cause warning
            output = network(source_img).clamp_(0, 1)  # we change this to [0,1]?

        # 保存去雾结果图像
        vutils.save_image(output, f'./result/result_{idx:04d}.png')

        mse_loss = F.mse_loss(output, target_img, reduction='none').mean((1, 2, 3))
        psnr448 = 10 * torch.log10(1 / mse_loss).mean()
        PSNR448.update(psnr448.item(), source_img.size(0))

        down_ratio = max(1, round(min(H, W) / 256))
        ssim_val = ssim(F.adaptive_avg_pool2d(output, (int(H / down_ratio), int(W / down_ratio))),
                        F.adaptive_avg_pool2d(target_img, (int(H / down_ratio), int(W / down_ratio))),
                        data_range=1, size_average=False).mean()
        SSIM_full.update(ssim_val.item(), source_img.size(0))

        print(PSNR448.avg, SSIM_full.avg)

    return PSNR448.avg, SSIM_full.avg


if __name__ == '__main__':
    device_index = [0]
    network = eval(args.model.replace('-', '_'))()
    network = nn.DataParallel(network, device_ids=device_index).cuda()
    network.load_state_dict(torch.load('D:/python_code/SFHformer-main/dehazing/sfhformer_haze_its.pth')['state_dict'])
    test_data_dir = 'E:/image_restoration/SOTS/indoor/'
    test_dataset = TestData_its(8, test_data_dir)
    test_loader = DataLoader(test_dataset,
                             batch_size=1,
                             shuffle=True,
                             num_workers=args.num_workers,
                             pin_memory=True)

    avg_psnr, avg_ssim = valid(test_loader, network)
    print(avg_psnr, avg_ssim)

**Comment by jxy33zrhd** [comment]
![Image](https://github.com/user-attachments/assets/3ea46a9e-954c-4a34-89a8-ba53e718b0a9)

Hello 您也可以检查一下这个地方 在所上传sfhformer_haze.py的第291行中 测试ITS的时候 需要将291行注释 测试OTS的时候 需要将291行激活

**Comment by Wjc369** [comment]
您好我已经找到问题，在sfhformer_haze.py的第291行中 测试ITS的时候 我没有进行注释，没想到结果差了这么多

**Comment by Wjc369** [comment]
非常感谢您的解答

**Comment by jxy33zrhd** [comment]
嗯嗯 没事 有问题随时提出
